lua/telescope/pickers.lua

```diff
@@ -452,21 +453,28 @@ function Picker:find()
 
     local process_complete = function()
       -- TODO: We should either: always leave one result or make sure we actually clean up the results when nothing matches
-      if selection_strategy == 'row' then
-        self:set_selection(self:get_selection_row())
-      elseif selection_strategy == 'follow' then
-        local index = self.manager:find_entry(self:get_selection())
-
-        if index then
-          local follow_row = self:get_row(index)
-          self:set_selection(follow_row)
-        else
+      if self.default_selection_index == nil then
+        if selection_strategy == 'row' then
+          self:set_selection(self:get_selection_row())
+        elseif selection_strategy == 'follow' then
+          local index = self.manager:find_entry(self:get_selection())
+
+          if index then
+            local follow_row = self:get_row(index)
+            self:set_selection(follow_row)
+          else
+            self:set_selection(self:get_reset_row())
+          end
+        elseif selection_strategy == 'reset' then
           self:set_selection(self:get_reset_row())
+        else
+          error('Unknown selection strategy: ' .. selection_strategy)
         end
-      elseif selection_strategy == 'reset' then
-        self:set_selection(self:get_reset_row())
       else
-        error('Unknown selection strategy: ' .. selection_strategy)
+        local idx = self.default_selection_index - 1
+        local base_row = self:get_reset_row()
+        local row = self.sorting_strategy == 'ascending' and (base_row + idx) or (base_row - idx)
+        self:set_selection(row)
```

Comment:
We should only do this when we open the Picker. When we type we should not go here but rather follow the selection_strategy, otherwise `_selection_row` could exceed the maximum displayed results.